streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -144,10 +148,15 @@
     private static final long DEFAULT_COMMIT_INTERVAL_MS = 30000L;
     private static final long EOS_DEFAULT_COMMIT_INTERVAL_MS = 100L;
     private static final int DEFAULT_TRANSACTION_TIMEOUT = 10000;
+    private static final short DEFAULT_MAX_CLIENT_TAG_KEY_VALUE_LENGTH = 40;
 
     public static final int DUMMY_THREAD_INDEX = 1;
     public static final long MAX_TASK_IDLE_MS_DISABLED = -1;
 
+    public static final int MAX_RACK_AWARE_ASSIGNMENT_TAG_LIST_SIZE = 10;
+    public static final int MAX_RACK_AWARE_ASSIGNMENT_TAG_KEY_LENGTH = 50;
+    public static final int MAX_RACK_AWARE_ASSIGNMENT_TAG_VALUE_LENGTH = 100;
```

Comment:
Do not have any since behind those restrictions, but thought it's better to have something.

Reply:
Actually, I don't think we need this limitation. Could we not be able to support list size > 10, key len > 50, value len > 100? I guess we can. If so, why this limitation? And like what you've asked, why this number?

Reply:
> Actually, I don't think we need this limitation. Could we not be able to support list size > 10, key len > 50, value len > 100?

I think we need some limitations because this information will be encoded into the subscription info and subscription info is the part of the group metadata message that we need to write into the internal topic. This was also discussed in the voting thread.

As for why those numbers, I really can't say. It can be more, can be less. But we need some limitation I think. But I can't imagine valid use case of having more than 10 tags with more than 50 and 100 characters length.

Reply:
Make sense to me. Could we add a comment above these line, to mention we have these limitation because these tag value will be encoded into subscription info, sth like that? Thanks.

Reply:
Thanks for the comment. I will add the integration test to test all three of those configurations.